### PR TITLE
ref(replay): Rewrite useReplayData to leverage useApiQuery 1/3

### PR DESCRIPTION
**Pull request by ryan953**
Per the title, this is a rewrite of useReplayData that leverages useApiQuery and friends to cache data, which will make navigations back/forward much faster. 

This part only addresses the first query: ``/organizations/${orgSlug}/replays/${replayId}/` which grabs the main replay record data. Navigating back & forth with be faster after this change, because this is a blocking fetch, but the data will already be in cache.

This is a slice from https://github.com/getsentry/sentry/pull/62109, to try to isolate test failures and keep the changes more readable (and not a big table-flip rewrite).

### CODE REVIEW

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 1]
```diff
@@ -22,7 +23,7 @@ type State = {
    */
   fetchingAttachments: boolean;
   fetchingErrors: boolean;
-  fetchingReplay: boolean;
+  // fetchingReplay: boolean;
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useReplayData.tsx:26]: Will there be no loading state at all?

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 2]
```diff
@@ -99,12 +100,25 @@ function useReplayData({
...
   const api = useApi();
+  const queryClient = useQueryClient();
 
   const [state, setState] = useState<State>(INITIAL_STATE);
   const [attachments, setAttachments] = useState<unknown[]>([]);
   const attachmentMap = useRef<Map<string, unknown[]>>(new Map()); // Map keys are always iterated by insertion order
   const [errors, setErrors] = useState<ReplayError[]>([]);
-  const [replayRecord, setReplayRecord] = useState<ReplayRecord>();
+
+  // Fetch every field of the replay. We're overfetching, not every field is used
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useReplayData.tsx:110]: This is just moved code, but might be helpful to include why we're fetching fields.

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 3]
```diff
@@ -213,21 +210,35 @@ function useReplayData({
...
+      });
+    };
+  }, [orgSlug, replayId, projectSlug, queryClient]);
+
   return {
     attachments,
     errors,
-    fetchError: state.fetchError,
-    fetching: state.fetchingAttachments || state.fetchingErrors || state.fetchingReplay,
-    onRetry: loadData,
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useReplayData.tsx:221]: So it looks like now we invalidate the cache on retry... will it also load the data afterwards?

Comment (ryan953) [reply static/app/utils/replays/hooks/useReplayData.tsx:221]: yes. When the cache is invalid, but the hook is still mounted, useQuery will re-fetch.

There are a few related options, like `refetchInterval`, `retryOnMount`, `retryDelay`, and `staleTime` that control fetching as well. None are really being leveraged here now though. https://tanstack.com/query/v4/docs/react/reference/useQuery

File: static/app/utils/replays/hooks/useReplayData.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/utils/replays/hooks/useReplayData.tsx:26]: There's still al loading state!
This value got moved out of the state that's manually tracked, and it's replaced by the return of `useApiQuery`.

`const {... , error: fetchReplayError} = useApiQuery(...)` is the new name for the same value, down a bit lower.

Comment (billyvg) [reply static/app/utils/replays/hooks/useReplayData.tsx:26]: Let's remove the commented code

**Review by billyvg — APPROVED** [review]

File: static/app/utils/replays/hooks/useReplayData.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/utils/replays/hooks/useReplayData.tsx:26]: d'oh

**Review by billyvg — APPROVED** [review]
Replays seem to still work ✅

**Pull request merged**